testcases/OsSecureBoot.py

```diff
@@ -0,0 +1,205 @@
+
+        con = self.cv_SYSTEM.console
+        con.run_command("test ! -f /sys/firmware/devicetree/base/ibm,secureboot/os-secureboot-enforcing")
+        con.run_command("test -f /sys/firmware/devicetree/base/ibm,secureboot/physical-presence-asserted")
+        con.run_command("test -f /sys/firmware/devicetree/base/ibm,secureboot/clear-os-keys")
+
+        for k in ["PK", "KEK", "db", "dbx"]:
+            # No keys should be enrolled, so size should be ascii "0" for each
+            output = con.run_command("cat /sys/firmware/secvar/vars/{}/size".format(k))
+            self.assertTrue("0" in output)
+
+            # Data should not contain anything
+            output = con.run_command("cat /sys/firmware/secvar/vars/{}/data | wc -c".format(k))
+            self.assertTrue("0" in output)
+
+
+    def addSecureBootKeys(self):
+        self.cv_SYSTEM.goto_state(OpSystemState.PETITBOOT_SHELL)
+        con = self.cv_SYSTEM.console
+
+        self.getTestData()
+
+        for k in ["PK", "KEK", "db", "dbx"]:
+            con.run_command("cat /tmp/{0}.auth > /sys/firmware/secvar/vars/{0}/update".format(k))
+
+        # System needs to power fully off to process keys on next reboot
+        self.cv_SYSTEM.goto_state(OpSystemState.OFF)  
+        self.cv_SYSTEM.goto_state(OpSystemState.PETITBOOT_SHELL)
+
+        con.run_command("test -f /sys/firmware/devicetree/base/ibm,secureboot/os-secureboot-enforcing")
+
+        for k in ["PK", "KEK", "db", "dbx"]:
+            # Size should return a nonzero ascii value when enrolled
+            output = con.run_command("cat /sys/firmware/secvar/vars/{}/size".format(k))
+            self.assertFalse("0" in output)
+
+            # Data should contain something
+            output = con.run_command("cat /sys/firmware/secvar/vars/{}/data | wc -c".format(k))
+            self.assertFalse("0" in output)
+
+
+    def checkKexecKernels(self):
+        self.cv_SYSTEM.goto_state(OpSystemState.PETITBOOT_SHELL)
+        con = self.cv_SYSTEM.console
+
+        self.getTestData(data="kernels")
+
+        # Fail regular kexec_load
+        output = con.run_command_ignore_fail("kexec -l /tmp/kernel-unsigned")
+        self.assertTrue("Permission denied" in "".join(output))
+
+        # Fail unsigned kernel
+        output = con.run_command_ignore_fail("kexec -s /tmp/kernel-unsigned")
+        self.assertTrue("Permission denied" in "".join(output))
+
+        # Fail dbx kernel
+        output = con.run_command_ignore_fail("kexec -s /tmp/kernel-dbx")
+        self.assertTrue("Permission denied" in "".join(output))
+
+        # Fail signed kernel with unenrolled key
+        output = con.run_command_ignore_fail("kexec -s /tmp/kernel-unenrolled")
+        self.assertTrue("Permission denied" in "".join(output))
+        
+        # Succeed good kernel
+        output = con.run_command("kexec -s /tmp/kernel-signed")
+        self.assertFalse("Permission denied" in "".join(output)) # may not be needed, command should fail
```

Comment:
Removed, `.run_command()` should raise an exception which serves the same purpose.

Reply:
> If you've been told by the HB team that this is fine then that needs to be documented here. Preferably along with reasons why that assumption is valid and when it might break (e.g. is this all going to fall over for the DT based attributes we have on Rainier?)

Added some additional comments, though I'm not 100% on the actual details. Will contact the people who know better and update with a better quote if needed.

> Skiboot tests are for doing unit testing inside of skiboot while op-test is for doing integration testing of the whole stack. Having tests in skiboot isn't a substitute. I'm not asking for super in-depth testing, just something to verify it actually works end to end.

Can you elaborate on what kind of test you are looking for? The keys are enrolled in an order that requires them to be validated. Any failures would result in empty variables, which is currently checked.

I'm adding a check on the `update-status` device tree entry which contains the update processing return code, though not currently included in this update as I'm debugging a potentially related issue.